src/main/java/com/parasoft/demoapp/grpc/service/JsonServiceImpl.java

```diff
@@ -1,9 +1,30 @@
 package com.parasoft.demoapp.grpc.service;
 
 
+import com.parasoft.demoapp.exception.ParameterException;
+import com.parasoft.demoapp.service.ItemInventoryService;
+import io.grpc.Status;
+import io.grpc.stub.StreamObserver;
 import net.devh.boot.grpc.server.service.GrpcService;
+import org.springframework.beans.factory.annotation.Autowired;
 
 @GrpcService
 public class JsonServiceImpl extends JsonServiceImplBase {
+    @Autowired
+    private ItemInventoryService itemInventoryService;
 
+    @Override
+    public void getStockByItemId(Long itemId, StreamObserver<Integer> responseObserver) {
+        try {
+            Integer inStock = itemInventoryService.getInStockByItemId(itemId);
+            responseObserver.onNext(inStock);
+            responseObserver.onCompleted();
+        } catch (ParameterException e) {
+            responseObserver.onError(Status.UNKNOWN
+                    .withDescription(e.getMessage())
+                    .withCause(e)
+                    .asRuntimeException());
+            e.printStackTrace();
```

Comment:
A comment out of scope of this task for @rwang-parasoft:
Why are we always printing the stack trace? I've noticed this in some other places before. I think normally this is only used when the coder needs to debug. It should not be included in normal log in my opinion.

Reply:
I aggree with you @yshi-parasoft, if the exception is handled I think we should not print the stack but sometimes we may need to print the stack if we swallow the exception on purpose(it is rare). In current situation I think it's no need to print the stack.